nupic/research/frameworks/vernon/mixins/oml.py

```diff
@@ -123,21 +155,53 @@ def create_test_test_dataloader(cls, config, dataset):
             pin_memory=torch.cuda.is_available(),
         )
 
+    def pre_task(self, tasks):
+        """Re-initialize task params prior to meta-train training."""
+        super().pre_task(tasks)
+        if not self.reset_task_params:
+            return
+
+        output_params = self.get_named_output_params().values()
+        for p in output_params:
+            if p.dim() == 2:
+                for t in tasks:
+                    task_weights = p[t, :].unsqueeze(0)
+                    nn.init.kaiming_normal_(task_weights)
+
     def post_epoch(self):
         super().post_epoch()
-        if self.should_stop() and self.run_meta_test:
-            # meta-training phase complete, perform meta-testing phase
-            for num_classes_learned in [10, 50, 100, 200, 600]:
-                if num_classes_learned > self.num_classes_eval:
+        if self.current_epoch >= self.epochs - 1 and self.run_meta_test:
```

Comment:
If self.current_epoch is zero indexed, isn't self.epochs -1 the last epoch? If that is the case == makes more sense than >=